src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -1046,6 +1058,78 @@ .maxstack  3
         }
 
         [Fact]
+        public void ValueTupleReturnMissingMemberWithCSharp7()
+        {
+            string source = @"
+class C
+{
+    public void M()
+    {
+        int x, y;
+        var nested = ((x, y) = (1, 2));
+        System.Console.Write(nested.x);
+    }
+}
+";
+
+            var comp = CreateCompilationWithMscorlib(source, references: new[] { ValueTupleRef, SystemRuntimeFacadeRef },
+                parseOptions: TestOptions.Regular.WithLanguageVersion(LanguageVersion.CSharp7));
+            comp.VerifyDiagnostics(
+                // (8,37): error CS8305: Tuple type '(int, int)' does not have an explicitly named element 'x'. Please use language version 7.1 or greater to access a unnamed element by its inferred name.
+                //         System.Console.Write(nested.x);
+                Diagnostic(ErrorCode.ERR_TupleInferredNamesNotAvailable, "x").WithArguments("(int, int)", "x", "7.1").WithLocation(8, 37)
+                );
+        }
+
+        [Fact]
+        public void ValueTupleReturnWithInferredNamesWithCSharp7_1()
+        {
+            string source = @"
+class C
+{
+    public void M()
+    {
+        int x, y, Item1, Rest;
+        var a = ((x, y) = (1, 2));
+        var b = ((x, x) = (1, 2));
+        var c = ((_, x) = (1, 2));
+        var d = ((Item1, Rest) = (1, 2));
+        var nested = ((x, Item1, y, (_, x, x), (x, y)) = (1, 2, 3, (4, 5, 6), (7, 8)));
+        (int, int) f = ((x, y) = (1, 2));
+    }
+}
+";
+            Action<ModuleSymbol> validator = module =>
+            {
+                var sourceModule = (SourceModuleSymbol)module;
+                var compilation = sourceModule.DeclaringCompilation;
+                var tree = compilation.SyntaxTrees.First();
+                var model = compilation.GetSemanticModel(tree);
+                var nodes = tree.GetCompilationUnitRoot().DescendantNodes();
+
+                var a = nodes.OfType<VariableDeclaratorSyntax>().ElementAt(4);
```

Comment:
Consider extracting a local for `nodes.OfType<VariableDeclaratorSyntax>()`. Perhaps with `.ToArray()`.